### DISCUSSION TITLE
Is there a way to directly compile c# into wasm?

### DISCUSSION THREAD

**Discussion by GreenShadeZhang** [dotnet/runtime#66446] [Q&A]
I'm doing istio-related work recently. I'm relatively familiar with c#. I just realized that it supports wasm when implementing the istio plugin, and the community has less documentation on compiling c# into wasm, so I want to ask if this is feasible? Or is the community interested in investing in doing something here?

The issues involved are at the link below
https://github.com/proxy-wasm/spec/issues/29

**Comment by hez2010** [comment]
See https://github.com/dotnet/runtimelab/tree/feature/NativeAOT-LLVM

**Reply by GreenShadeZhang** [reply]
> See https://github.com/dotnet/runtimelab/tree/feature/NativeAOT-LLVM

thank you very much.
